### PR TITLE
New workflow client update

### PR DESCRIPTION
**Pull request by mcalmer**
# Description

Added a new workflow which describe how to update clients using recurring actions.

# Target branches

The described states are available with 4.3.12

Backport targets (edit as needed):

- [x] master
- [x] 4.3 https://github.com/uyuni-project/uyuni-docs/pull/2927

# Links
- Issue: https://github.com/SUSE/spacewalk/issues/23573

### CODE REVIEW

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 1]
```diff
@@ -0,0 +1,88 @@
+[[workflow-client-update]]
+
+= Update Clients
+
+This workflow show how to automate update clients registered at {productname} using recurring actions.
+
+The workflow is applicable for Salt clients.
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:7]:
```suggestion

// FIXME uncomment on 4.3 backport
// The workflow is applicable for Salt clients.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 2]
```diff
@@ -0,0 +1,88 @@
+[[workflow-client-update]]
+
+= Update Clients
+
+This workflow show how to automate update clients registered at {productname} using recurring actions.
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:5]:
```suggestion
This workflow shows how to automate update clients registered at {productname} using recurring actions.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+The workflow is applicable for Salt clients.
+
+
+== Use case / Situation
+
+Automated update of clients is benefitial when:
+
+- update of a large number of clients is wanted
+- the workflow should not be re-done every execution
+- a fix maintenance window exists
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:16]:
```suggestion
- a dedicated maintenance window exists
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+== Outcome / Resolution
+
+Successful completion of this workflow results in consistent, supportable outcomes.
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:20]:
```suggestion
Successful completion of this workflow results in consistent and supportable state.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+
+== Preparation
+
+Before you start, you should have a number of salt clients onboarded.
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:25]:
```suggestion
// FIXME 4.3 backport wersion
// Before you start, you should have a number of {salt} clients onboarded.
Before you start, you should have a number of clients onboarded.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+It may make sense to have them sorted into groups you want to update together.
+In this workflow we use a system group named "infra-services".
+
+
+== Step-by-step workflow instructions
+
+To update a salt client 2 steps are required. A third step is optional but is highly recommended to finalize the update process.
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:32]:
```suggestion
To update a client two steps are required.
A third step is optional but it is highly recommended to finalize the update process.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+Procedure 1: Create a recurring action to update salt itself:
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:34]:
```suggestion
Procedure 1: Create a recurring action to update {salt} itself
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+[role=procedure]
+. As an example we create the action to update salt itself as a recurring action for all systems in the organization.
+In the {productname} Web UI go to Home > My Organization > Recurring Actions and click `Create`.
+. Select `Action Type` "Custom State" and enter a `Schedule Name` like "update-salt"
+. Select a Schedule. E.g. `Weekly`: Wednesday, 9:00 am
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:39]:
```suggestion
. Select a schedule.
  For example, `Weekly`: Wednesday, 9:00 am.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+. Assign the `update-salt` state by selecting the checkbox
+. Save the action by clicking on `Save Changes`
+. You can edit the execution order of the states if needed. Click on `Confirm` to agree on the order.
+. To save the action, click on `Create Schedule`.
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:43]:
```suggestion
. Assign the `update-salt` state by selecting the checkbox.
. Save the action by clicking btn:[Save Changes].
. You can edit the execution order of the states if needed.
  Click btn:[Confirm] to confirm the order.
. Click btn:[Create Schedule] to save the action.
```

**Review by keichwa — CHANGES_REQUESTED** [review]
Thanks for the draft!  I'll continue with the review later.

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+Procedure 2: Create a recurring action to apply all available updates to the systems:
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:45]:
```suggestion
.Procedure 2: Creating a recurring action to apply all available updates to the systems
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+[role=procedure]
+. As an example we create the action to apply all updates as a recurring action for a system group called "infra-services".
+In the {productname} Web UI go to Systems > System Groups and click on "infra-services".
```
Comment (keichwa) [inline modules/common-workflows/pages/workflow-client-update.adoc:48]:
```suggestion
. As an example, create an action to apply all updates as a recurring action for a system group called "infra-services".
  In the {productname} Web UI go to Systems > System Groups and click btn:[infra-services].
```

**Review by keichwa — CHANGES_REQUESTED** [review]
cont.

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 12]
```diff
@@ -0,0 +1,88 @@
+[[workflow-client-update]]
+
+= Update Clients
+
+This workflow shows how to automate update clients registered at {productname} using recurring actions.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:5]:
```suggestion
This workflow shows how to automate updating the clients registered at {productname} using recurring actions.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+
+The workflow is applicable for Salt clients.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:7]:
```suggestion
The workflow is applicable for {salt} clients.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+
+
+== Use case / Situation
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:10]:
```suggestion
== Use Case / Situation
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+- a dedicated maintenance window exists
+
+== Outcome / Resolution
+
+Successful completion of this workflow results in consistent and supportable state.
+
+
+== Preparation
+
+Before you start, you should have a number of salt clients onboarded.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:25]:
```suggestion
Before you start, you should have a number of {salt} clients onboarded.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+It may make sense to have them sorted into groups you want to update together.
+In this workflow we use a system group named "infra-services".
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:27]:
```suggestion
In this workflow we use a system group named [literal]``infra-services``.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+
+
+== Step-by-step workflow instructions
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:30]:
```suggestion
== Step-by-step Workflow Instructions
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+
+To update a client two steps are required.
+A third step is optional but it is highly recommended to finalize the update process.
+
+.Procedure 1: Creating a recurring action to update {salt} itself
+[role=procedure]
+. As an example we create the action to update salt itself as a recurring action for all systems in the organization.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:37]:
```suggestion
. As an example we create the action to update {salt} itself as a recurring action for all systems in the organization.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+In the {productname} Web UI go to Home > My Organization > Recurring Actions and click `Create`.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:38]:
```suggestion
In the {productname} {webui} go to menu:Home[My Organization > Recurring Actions] and click btn:[Create].
```

**Review by 0rnela — APPROVED** [review]

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+. To save the action, click on `Create Schedule`.
+
+The `uptodate` state perform also a reboot, if an update request this. This can be prevented if wanted. For example at systems which use live patching.
+
+Procedure 3: Configure systems to not reboot when apply the `uptodate` state:
+[role=procedure]
+. In the {productname} Web UI go to Systems > Custom System Info and click `Create Key`.
+. Enter "mgr_reboot_if_needed" as `Key Label` and set as `Description` 
+'Define if the up2date state should perform a reboot if needed. Set to "false" if this is not wanted'
+. Click on `Create Key` to store the new key.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:75]:
```suggestion
. Click on btn:[Create Key] to store the new key.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+. To assign values to multiple systems, go in the {productname} Web UI to Systems > Overview and select the checkbox for all systems you want to modify.
+These system are not in the System Set Manager (SSM)
+. Select in the left menu System Set Manager and select `Misc > Custom Values` in the tab bar.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:78]:
```suggestion
. Select in the left menu System Set Manager and select menu:Misc[Custom Values] in the tab bar.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+. Click on `mgr_reboot_if_needed` and enter "false" as `Value`.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:79]:
```suggestion
. Click on [literal]``mgr_reboot_if_needed`` and enter **False** as [literal]``Value``.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+. Click on `Set Values` to save.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:80]:
```suggestion
. Click btn:[`Set Values] to save.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:81]:
```suggestion


```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 25]
```diff
@@ -0,0 +1,95 @@
...
+Procedure 4: Create a recurring action to run a highstate after the update:
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:82]:
```suggestion
Procedure 4: Creating a Recurring Action to Run a Highstate After the Update
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 26]
```diff
@@ -0,0 +1,95 @@
...
+. As an example we create the action to apply the highstate for the same group which was fully updated before.
+In the {productname} Web UI go to Systems > System Groups and click on "infra-services".
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:84]:
```suggestion
. As an example we create the action to apply the highstate for the same group which was fully updated before.
  In the {productname} {webui} go to menu:Systems[System Groups] and click on [literal]``infra-services``.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 27]
```diff
@@ -0,0 +1,95 @@
...
+. Now go to `Recurring Actions` and click on `Create`.
+. Select `Action Type` "Highstate" and enter a `Schedule Name` like "highstate"
+. Select a Schedule. E.g. `Weekly`: Wednesday, 10:30 am.
+Again keep enough time between this action and the "full-system-update" action.
+. To save the action, click on `Create Schedule`.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:89]:
```suggestion
. Click btn:[Create Schedule] to save the action..
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 28]
```diff
@@ -0,0 +1,95 @@
...
+. Click on `mgr_reboot_if_needed` and enter "false" as `Value`.
+. Click on `Set Values` to save.
+
+Procedure 4: Create a recurring action to run a highstate after the update:
+. As an example we create the action to apply the highstate for the same group which was fully updated before.
+In the {productname} Web UI go to Systems > System Groups and click on "infra-services".
+. Now go to `Recurring Actions` and click on `Create`.
+. Select `Action Type` "Highstate" and enter a `Schedule Name` like "highstate"
+. Select a Schedule. E.g. `Weekly`: Wednesday, 10:30 am.
+Again keep enough time between this action and the "full-system-update" action.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:88]:
```suggestion
  Again, keep enough time between this action and the [literal]``full-system-update`` action.
```

File: modules/common-workflows/pages/workflow-client-update.adoc [hunk 29]
```diff
@@ -0,0 +1,95 @@
...
+. Select in the left menu System Set Manager and select `Misc > Custom Values` in the tab bar.
+. Click on `mgr_reboot_if_needed` and enter "false" as `Value`.
+. Click on `Set Values` to save.
+
+Procedure 4: Create a recurring action to run a highstate after the update:
+. As an example we create the action to apply the highstate for the same group which was fully updated before.
+In the {productname} Web UI go to Systems > System Groups and click on "infra-services".
+. Now go to `Recurring Actions` and click on `Create`.
+. Select `Action Type` "Highstate" and enter a `Schedule Name` like "highstate"
+. Select a Schedule. E.g. `Weekly`: Wednesday, 10:30 am.
```
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:87]:
```suggestion
. Select a Schedule. E.g. `Weekly`: Wednesday, 10:30 am.
For example, ** Weekly: Wednesday, 10:30 am **.
```

File: modules/common-workflows/pages/workflow-client-update.adoc (hunk 28 above)
Comment (0rnela) [inline modules/common-workflows/pages/workflow-client-update.adoc:88]:
```suggestion
Again, keep enough time between this action and the [literal]``full-system-update`` action.
```

**Pull request merged**